<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CatWindow</class>
 <widget class="QMainWindow" name="CatWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>737</width>
    <height>692</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CatWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>721</width>
      <height>621</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QGraphicsView" name="graphicsView">
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="sceneRect">
            <rectf>
             <x>0.000000000000000</x>
             <y>0.000000000000000</y>
             <width>500.000000000000000</width>
             <height>500.000000000000000</height>
            </rectf>
           </property>
           <property name="renderHints">
            <set>QPainter::Antialiasing|QPainter::TextAntialiasing</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="vizSelectBox">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Choose the visualization type.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Catalan Path: A sequence of moves either East or South on a grid, &lt;/p&gt;&lt;p&gt;forming a path that never dips below the main diagonal. &lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Polygon Triangulation: Lines forming a triangulation of a regular polygon. An&lt;/p&gt;&lt;p&gt;&amp;quot;East&amp;quot; move changes the focus vertex (circled in gray), and a &amp;quot;South&amp;quot; move&lt;/p&gt;&lt;p&gt;(except for the very last one) makes a triangle from the focus vertex to the&lt;/p&gt;&lt;p&gt;nearest available vertex.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <item>
            <property name="text">
             <string>Catalan Path</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Polygon Triangulation</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="eastButton">
             <property name="toolTip">
              <string>Move east.</string>
             </property>
             <property name="text">
              <string>East</string>
             </property>
             <property name="shortcut">
              <string>Right</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="southButton">
             <property name="toolTip">
              <string>Move south.</string>
             </property>
             <property name="text">
              <string>South</string>
             </property>
             <property name="shortcut">
              <string>Down</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="undoButton">
           <property name="toolTip">
            <string>Undo last move.</string>
           </property>
           <property name="text">
            <string>Undo</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="clearButton">
           <property name="toolTip">
            <string>Clear the board.</string>
           </property>
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="randButton">
           <property name="text">
            <string>Random</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="dimLabel">
             <property name="focusPolicy">
              <enum>Qt::ClickFocus</enum>
             </property>
             <property name="text">
              <string>Dimension (2-20)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="dimensionBox">
             <property name="focusPolicy">
              <enum>Qt::StrongFocus</enum>
             </property>
             <property name="toolTip">
              <string>Set the board dimension (2 - 20)</string>
             </property>
             <property name="keyboardTracking">
              <bool>false</bool>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>20</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="resizeButton">
           <property name="toolTip">
            <string>Resize the board to the value in the Dimension box.</string>
           </property>
           <property name="text">
            <string>Resize</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="dyckWord">
           <property name="font">
            <font>
             <pointsize>18</pointsize>
            </font>
           </property>
           <property name="focusPolicy">
            <enum>Qt::ClickFocus</enum>
           </property>
           <property name="text">
            <string>(Empty path)</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignHCenter|Qt::AlignTop</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="quitButton">
           <property name="text">
            <string>Quit</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QProgressBar" name="eastBar">
       <property name="maximum">
        <number>5</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="format">
        <string>%v / %m East Steps</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QProgressBar" name="southBar">
       <property name="maximum">
        <number>5</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="format">
        <string>%v / %m South Steps</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>737</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
